src/plugins/accounting.cpp

```diff
@@ -35,3 +35,67 @@ Association* get_association (int userid,
 
     return &bank_it->second;
 }
+
+
+std::string Association::to_json () const
+{
+    json_t *u = json_object ();
+
+    json_object_set_new (u, "bank_name", json_string (bank_name.c_str ()));
+    json_object_set_new (u, "fairshare", json_real (fairshare));
+    json_object_set_new (u, "max_run_jobs", json_integer (max_run_jobs));
+    json_object_set_new (u, "cur_run_jobs", json_integer (cur_run_jobs));
+    json_object_set_new (u, "max_active_jobs", json_integer (max_active_jobs));
+    json_object_set_new (u, "cur_active_jobs", json_integer (cur_active_jobs));
+
+    json_t *held_job_ids = json_array ();
+    for (const auto &job_id : held_jobs) {
+        json_array_append_new (held_job_ids, json_integer (job_id));
+    }
+    json_object_set_new (u, "held_jobs", held_job_ids);
+
+    json_t *user_queues = json_array ();
+    for (const auto &queue : queues) {
+        json_array_append_new (user_queues, json_string (queue.c_str ()));
+    }
+    json_object_set_new (u, "queues", user_queues);
+
+    json_object_set_new (u, "queue_factor", json_integer (queue_factor));
+    json_object_set_new (u, "active", json_integer (active));
+
+    char *json_str = json_dumps (u, JSON_INDENT (4));
+    std::string result (json_str);
+    free (json_str);
+    json_decref (u);
+
+    return result;
```

Comment:
There's no error checking in this function. `json_object` for instance can return `NULL` and so can e.g. `json_integer`. I'm not sure what exactly would happen though...

Could you maybe create `held_job_ids` and `user_queues`, checking if either creation fails, add all the elements to them (checking for errors as you go) and then pack everything into `u` with a `json_pack`? You could have the pack steal the references to `held_job_ids` and `user_queues` too. That would combine all your error checking into three locations. Plus one at the end to check that the `json_dumps` succeeds?

Reply:
Oh, sure, that's a great suggestion, thanks. Yeah, I think what I could do is move the creation of both `json_array ()`s first (and check for errors upon creation there), and _then_ pack everything all at once into `u` with `json_pack ()`. I'll fix this 👍